### ISSUE TITLE
Linking warnings after optimizer when calling Short.shortValue()

### ISSUE THREAD

**Issue by sjrd** [scala-js/scala-js#2780]
The following snippet:
```scala
    val x = new java.lang.Short(5.toShort)
    println(x.shortValue())
```
causes *warnings* in the refiner (after optimization), saying:
```
[warn] Referring to non-existent method java.lang.Integer.shortValue()scala.Short
[warn]   called from helloworld.HelloWorld$.main()scala.Unit
[warn]   called from core module entry points
[warn] involving instantiated classes:
[warn]   helloworld.HelloWorld$
```
This is because the first linker pass discards `java.lang.Integer.shortValue()`, as it is unused. But then the optimizer transforms the above into:
```scala
val x: int = 5
println(x.shortValue())
```
thanks to box/unbox elimination. And then it tries to invoke `shortValue()` on a primitive `int`, which makes it look into `Integer` instead of `Short`.

This does not prevent the program to be well-defined and work, so it's not super urgent to fix. Besides, it will be fixed by construction if we add primitive `byte`s and `short`s in the IR, which I intend to do in 1.0.0.

**Comment by sjrd** [comment]
It turns out that there is also an easy fix in 0.6.x. See #2868.

**Comment by sjrd** [comment]
Fixed in #2868.
